plugins/functions/string.c

```diff
@@ -294,6 +357,198 @@ func_string_tokenize(grn_ctx *ctx, int n_args, grn_obj **args,
+
+    GRN_TEXT_INIT(&inspected, 0);
+    grn_inspect(ctx, &inspected, nth_or_name);
+    GRN_PLUGIN_ERROR(ctx, GRN_INVALID_ARGUMENT,
+                     "[string_slice][nth_or_name] must be a text or number bulk: %.*s",
+                     (int)GRN_TEXT_LEN(&inspected),
+                     GRN_TEXT_VALUE(&inspected));
+    GRN_OBJ_FIN(ctx, &inspected);
+    return NULL;
+  }
+
+  //TODO: should cache
+  OnigRegex regex = grn_onigmo_new(ctx,
+                                   GRN_TEXT_VALUE(pattern),
+                                   GRN_TEXT_LEN(pattern),
+                                   GRN_ONIGMO_OPTION_DEFAULT,
+                                   GRN_ONIGMO_SYNTAX_DEFAULT,
+                                   "[string_slice][regexp]");
+
+  if (!regex) {
+    return NULL;
+  }
+
+  const char *target = GRN_TEXT_VALUE(target_raw);
+  int64_t target_length = GRN_TEXT_LEN(target_raw);
+
+  OnigRegion region;
+  onig_region_init(&region);
+
+  //Cannot use normalized string. 
+  //The matching parts of original string cannot be inferred from the matching parts of the normalized string.
+  OnigPosition position = onig_search(regex,
+                                      target,
+                                      target + target_length,
+                                      target,
+                                      target + target_length,
+                                      &region,
+                                      ONIG_OPTION_NONE);
+
+  if (position != ONIG_MISMATCH) {
+    int64_t nth = -1;
+
+    if (grn_obj_is_text_family_bulk(ctx, nth_or_name)) {
+      const char *name = GRN_TEXT_VALUE(nth_or_name);
+      int64_t name_length = GRN_TEXT_LEN(nth_or_name);
+
+      nth = onig_name_to_backref_number(regex,
+                                        name,
+                                        name + name_length,
+                                        &region);
+
+    } else if (grn_obj_is_number_family_bulk(ctx, nth_or_name)) {
+      nth = grn_plugin_proc_get_value_int64(ctx,
+                                            nth_or_name,
+                                            0,
+                                            "[string_slice][capture_num]");
+    }
+
+    if (nth >= 0 && nth < region.num_regs) {
+      OnigPosition start = region.beg[nth];
+      OnigPosition end = region.end[nth];
+
+      result = grn_plugin_proc_alloc(ctx, user_data, target_raw->header.domain, 0);
+      if (!result) {
+        goto exit_with_onig_free;
+      }
+      GRN_BULK_REWIND(result);
+      GRN_TEXT_SET(ctx, result, target + start, end - start);
+    }
+  }
+
+  if (!result) {
+    if (!default_value) {
+      default_value = grn_plugin_proc_alloc(ctx, user_data, target_raw->header.domain, 0);
+      if (!default_value) {
+        goto exit_with_onig_free;
+      }
+      GRN_BULK_REWIND(default_value);
+    }
+    result = default_value;
+  }
+
+exit_with_onig_free:
+
+  onig_region_free(&region, false);
```

Comment:
Could you use `0` not `false` here? Because `free_self` is declared as `int` not `bool`: `void onig_region_free(OnigRegion* region, int free_self);`

```suggestion
  onig_region_free(&region, 0);
```

Reply:
I fixed it.